CSharp/BatchExplorer/Service/BatchService.cs

```diff
@@ -201,13 +201,56 @@ public Task<CloudPool> GetPoolAsync(string poolId)
             int targetDedicated,
             TimeSpan? timeout,
             ComputeNodeDeallocationOption? deallocationOption)
-        {
+        {
+            var pool = await this.GetPoolAsync(poolId);
+            if (pool.AutoScaleEnabled.HasValue && pool.AutoScaleEnabled.Value)
+            {
+                await this.Client.PoolOperations.DisableAutoScaleAsync(poolId);
+                if (!await this.WaitForResizingOperationToFinish(poolId))
+                {
+                    throw new Exception("Autoscale disable operation is already in progress. Please try again after sometime to Resize the pool.");
+                }
+            }
+
             await this.Client.PoolOperations.ResizePoolAsync(poolId, targetDedicated, timeout, deallocationOption);
+        }
+
+        public async Task EnableAutoScale(string poolId, string autoScaleformula)
+        {
+            await this.Client.PoolOperations.EnableAutoScaleAsync(poolId, autoScaleformula);
         }
 
         public IPagedEnumerable<NodeAgentSku> ListNodeAgentSkus()
         {
             return this.Client.PoolOperations.ListNodeAgentSkus();
+        }
+
+        /// <summary>
+        /// Evaluates the autoscale formula.
+        /// </summary>
+        /// <param name="poolId">The pool identifier.</param>
+        /// <param name="autoScaleformula">The autoscale formula.</param>
+        /// <returns>Result of evaluation</returns>
+        public async Task<string> EvaluateAutoScaleFormula(string poolId, string autoScaleformula)
+        {
+            CloudPool pool = this.Client.PoolOperations.GetPool(poolId);
+            if (pool.AutoScaleEnabled.HasValue && pool.AutoScaleEnabled.Value)

```

Comment:
I don't think it's really required to get the pool and check if it is autoscale enabled before issuing the evaluate autoscale request -- if this operation fails, a popup box should trigger handling the exception thrown from the service, and that exception should contain details about not being able to evaluate autoscale on a pool which doesn't have autoscale enabled.

In short, I think something like this is better:
```
AutoScaleRun eval = await this.Client.PoolOperations.EvaluateAutoScaleAsync(pool.Id, autoScaleformula);

if (eval.Error == null)
{
    return eval.Results;
}
else
{                    
    return eval.Error.Message;
}
```